installer/pkg/config-generator/generator_test.go

```diff
@@ -11,19 +11,31 @@ import (
 	"github.com/openshift/installer/pkg/asset/tls"
 )
 
-func initConfig(t *testing.T, file string) ConfigGenerator {
+var pullSecretContents = []byte(`{"dummy":"value"}`)
+
+func initConfig(t *testing.T, file string) (ConfigGenerator, func()) {
 	cluster, err := config.ParseConfigFile("./fixtures/" + file)
 	if err != nil {
 		t.Errorf("Test case TestUrlFunctions: failed to parse test config, %s", err)
 	}
 
+	tf, err := ioutil.TempFile("", "generator_test")
+	if err != nil {
+		t.Errorf("error creating temp file: %v", err)
+	}
+	if _, err := tf.Write(pullSecretContents); err != nil {
+		t.Errorf("error writing to temp file %q: %v", tf.Name(), err)
```

Comment:
To clarify [my earlier comments][1], I don't think we want to store the pull secret path at all, and that we should replace `PullSecretPath` with a `PullSecret string` in the `Cluster` type.  Then there would be no need for a temporary file here.  The only filesystem connection would be on input, where we would provide users with the option to load the pull secret from a file.  I set up something similar for SSH pubkeys [here][2].  For pull secrets, I expect the logic to be something like:

```
Pull secret:
Enter your pull secret as a single line of JSON (e.g. '{...}') or as a path to a file containing the pull secret JSON.
```

Then use an "is this valid JSON?" on the result.  If it is, assume it's the pull secret.  If not, assume it's the path and try to load valid JSON from the path.  If you can't, complain about it and post the prompt again.

We have an existing "does this file contain valid JSON?" check [here][3], but I think you'll want something like [this][4] to make it easy to get good invalid-JSON error messages for both strings and files.

[1]: https://github.com/openshift/installer/pull/205/#discussion_r215117166
[2]: https://github.com/openshift/installer/pull/199/files#diff-0fa610c35b30a72dea1ff2a8c0b1e2e9R26
[3]: https://github.com/openshift/installer/blob/f2e6b3ed56aa114cd263369fd626d6b0e493e6ed/installer/pkg/validate/validate.go#L32
[4]: https://github.com/wking/openshift-installer/commit/bd1ebb3ac784c606dcc58aa42cc790513c36306b#diff-3a2d1fe927780885ccdacf5377759554L41

Reply:
@wking 
I'm sure we can change the `Cluster` type to store the pull secret. But i'm not convinced this the PR to change how we consume the pullsecret in `Cluster`

Reply:
> I'm sure we can change the `Cluster` type to store the pull secret. But i'm not convinced this the PR to change how we consume the pullsecret in `Cluster`

I've filed #208 with the file -> string change.  I'm fine landing that PR first and then rebasing this one on top (which would let you drop this temp file, etc.).  I'm also fine landing this PR first and rebasing mine on top.  Which way do you want to go?

Reply:
I can rebase on top of 208